### PR TITLE
En_Hidden_Nuts

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c [hunk 1]
```diff
@@ -36,51 +42,396 @@ const ActorInit En_Hidden_Nuts_InitVars = {
...
+    func_80BDB268(this);
+}
+
+void EnHiddenNuts_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHiddenNuts* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BDB1B4(EnHiddenNuts* this, s32 arg0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c:117]:
```suggestion
void func_80BDB1B4(EnHiddenNuts* this, s32 arg1) {
```

File: src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c [hunk 2]
```diff
@@ -36,51 +42,396 @@ const ActorInit En_Hidden_Nuts_InitVars = {
...
 
-#endif
+static AnimationHeader* D_80BDC0FC[] = {
+    &object_hintnuts_Anim_000168, &object_hintnuts_Anim_0024CC, &object_hintnuts_Anim_0026C4,
+    &object_hintnuts_Anim_002894, &object_hintnuts_Anim_002B90, &object_hintnuts_Anim_002F7C,
+    &object_hintnuts_Anim_003128, &object_hintnuts_Anim_0029BC, &object_hintnuts_Anim_0024CC,
+};
+
+u8 D_80BDC120[] = {
+    2, 2, 2, 0, 2, 0, 0, 2, 2, 0,
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c:72]:
```suggestion
    ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_LOOP, ANIMMODE_LOOP, ANIMMODE_ONCE, ANIMMODE_ONCE, ANIMMODE_LOOP,
```
Based on usage

File: src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c [hunk 3]
```diff
@@ -36,51 +42,396 @@ const ActorInit En_Hidden_Nuts_InitVars = {
...
+
+        for (i = 0; i < 3; i++) {
+            sp94.y = Rand_ZeroFloat(1.0f) + 1.0f;
+
+            Math_Vec3f_Copy(&sp74, &this->actor.world.pos);
+
+            sp74.x += randPlusMinusPoint5Scaled(15.0f);
+            sp74.z += randPlusMinusPoint5Scaled(15.0f);
+
+            EffectSsDtBubble_SpawnCustomColor(globalCtx, &sp74, &sp94, &sp88, &sp84, &sp80, Rand_S16Offset(120, 0x5A),
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c:177]:
```suggestion
            EffectSsDtBubble_SpawnCustomColor(globalCtx, &sp74, &sp94, &sp88, &sp84, &sp80, Rand_S16Offset(120, 90),
```

File: src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c [hunk 4]
```diff
@@ -36,51 +42,396 @@ const ActorInit En_Hidden_Nuts_InitVars = {
...
+
+        for (i = 0; i < 3; i++) {
+            sp98.y = Rand_ZeroFloat(1.0f) + 1.0f;
+
+            Math_Vec3f_Copy(&sp78, &this->actor.world.pos);
+
+            sp78.x += randPlusMinusPoint5Scaled(15.0f);
+            sp78.z += randPlusMinusPoint5Scaled(15.0f);
+
+            EffectSsDtBubble_SpawnCustomColor(globalCtx, &sp78, &sp98, &sp8C, &sp88, &sp84, Rand_S16Offset(120, 0x5A),
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c:220]:
```suggestion
            EffectSsDtBubble_SpawnCustomColor(globalCtx, &sp78, &sp98, &sp8C, &sp88, &sp84, Rand_S16Offset(120, 90),
```

File: src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c [hunk 5]
```diff
@@ -36,51 +42,396 @@ const ActorInit En_Hidden_Nuts_InitVars = {
...
+void func_80BDB7E8(EnHiddenNuts* this, GlobalContext* globalCtx) {
+    Vec3f sp3C;
+
+    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+        ActorCutscene_Stop(0x7C);
+        ActorCutscene_SetIntentToPlay(this->unk_226);
+        return;
+    }
+
+    if (ActorCutscene_GetCanPlayNext(this->unk_226) == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Hidden_Nuts/z_en_hidden_nuts.c:250]:
```suggestion
    if (!ActorCutscene_GetCanPlayNext(this->unk_226)) {
```

**Review by hensldm — COMMENTED** [review]
Some minor things.

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**